artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerSessionImpl.java

```diff
@@ -540,6 +541,12 @@ public ServerConsumer createConsumer(final long consumerID,
             props.putSimpleStringProperty(ManagementHelper.HDR_FILTERSTRING, filterString);
          }
 
+         String protocolName = remotingConnection.getProtocolName();
+         if (protocolName != null && protocolName.equals(ProtocolName.MQTT_PROTOCOL_NAME.getProtocolName().toString())) {
```

Comment:
This is protocol specific logic. This MUST not ever live in core modules. Protocol specific logic can ONLY reside in its protocol module